appendix/single-sign-on.rst

```diff
@@ -0,0 +1,521 @@
+* a ``REMOTE_USER`` web server environment variable
+* an ``HTTP_REMOTE_USER`` web server environment variable
+
+it creates a new session for that user.
+
+.. note:: 😬 **Wait. SSO allows you to sign in with only a username?**
+
+   In principle, yes.
+
+   **How is that okay?**
+
+   In this guide, we configure our web server (Apache)
+   to intercept all requests to the ``/auth/sso`` endpoint.
+   Instead of forwarding them to Zammad,
+   Apache initiates a three-sided login process (*Kerberos authentication*)
+   between the itself, the user, and the Active Directory server.
+
+   If Active Directory doesn’t recognize the user or their password,
+   Zammad never sees the request, and the session is never created.
+
+   **What does this all mean?**
+
+   It means there are many ways you could set up SSO—you
+   don’t need to follow this guide or even use Active Directory or Kerberos—but
+   if you don’t know what you’re doing,
+   you’re going to end up with a *massive* security hole.
+
+Getting Started
+---------------
+
+.. hint:: 😵 **Too busy to handle it on your own?**
+
+   We’ve got you covered.
+   Our experts offer custom-tailored workshops
+   to get your team up and running fast and with confidence.
+   `Just drop us a line <https://zammad.com/contact>`_!
+
+You will need:
+
+* a Microsoft Active Directory environment with
+
+  * root access
+  * support for AES 256-bit encryption
+
+* a Zammad host with
+
+  * root access
+  * a fully-qualified domain name (FQDN)
+
+* some familiarity with system administration (*e.g.,* Apache configuration)
+
+For best results, set up `LDAP integration`_
+to make sure your Active Directory and Zammad user accounts
+are always in sync.
+
+.. _LDAP integration: https://admin-docs.zammad.org/en/latest/system/integrations/ldap.html
+
+.. _sso-register-spn:
+
+Step 1: Configure Active Directory
+----------------------------------
+
+In the Kerberos authentication scheme,
+the **authentication server** (Active Directory)
+needs to maintain shared secrets with the **service** (Zammad).
+To make this possible, we need to register a **service principal name** (SPN)
+for Zammad on Active Directory.
+
+.. note:: These directions have been confirmed on Windows Server 2016.
+
+1a. Create a service account
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You may use an existing service account if you have one.
+Admin privileges are not required; a normal user account will do.
+
+Select “This account supports Kerberos AES 256 bit encryption” under
+**Properties > Account > Account options**.
+
+1b. Register an SPN for Zammad
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. note:: Replace the following placeholders in the command below:
+
+   :``<zammad-host>``:              Zammad FQDN
+   :``<service-acct>``:             Service account logon name
+   :``<service-acct-pwd>``:         Service account password
+   :``<domain>``:                   Windows domain
+   :``<master-domain-controller>``: Master domain controller IP/FQDN
```

Comment:
I used angle brackets (`<` / `>`) instead of curly braces (`{` / `}`) for all of these placeholders because the sample Apache vhost config uses actual curly braces:

```apache
RewriteCond %{LA-U:REMOTE_USER} (.+)
```

Reply:
Sure, good point!